core/config/strategies/uniStrategy.js

```diff
@@ -7,64 +7,58 @@ import {
   mustBeComparable,
 } from '../../../util/sameStructure';
 
-// A uniAssay makes uni amounts, which are either empty or have unique
-// descriptions. The quantity must either be null, in which case it is
-// empty, or be some truthy comparable value, in which case it
-// represents a single unique unit described by that truthy
-// quantity. Combining two uni amounts with different truthy
-// quantities fails, as they represent non-combinable rights.
-const makeUniStrategy = (descriptionCoercer = d => d) => {
+// The uniStrategy represents quantities that can never be combined.
+// For example, usually there is only one invite in an invite purse or
+// payment. (Using a listStrategy is an alternative, but when there is
+// usually a quantity of one, it's bothersome to always have to grab
+// the first item in the list rather than just represent the item
+// itself.)
+
+// The uni quantities are either empty (null) or unique. Combining two
+// non-null uni quantities fails because they represent non-combinable
+// rights. In a commonly used pattern, uni quantities contain an id
+// represented by a unique empty object.
+
+// `customInsistKind` enforces the particular kind of thing represented. For
+// example, the quantity in an invitation to join a contract might look like:
+// {
+//   id: harden({}),
+//   offerToBeMade: [rule1, rule2],
+// }
+
+const makeUniStrategy = customInsistKind => {
   const uniStrategy = harden({
-    insistKind: optDescription => {
-      if (optDescription === null) {
+    insistKind: quantity => {
+      if (quantity === null) {
         return null;
       }
-      insist(
-        !!optDescription,
-      )`Uni optDescription must be either null or truthy: ${optDescription}`;
-      mustBeComparable(optDescription);
-      const description = descriptionCoercer(optDescription);
-      insist(!!description)`Uni description must be truthy ${description}`;
-      mustBeComparable(description);
-      return description;
+      mustBeComparable(quantity);
+      return customInsistKind(quantity);
```

Comment:
The name `customInitKind` suggests that it just approves or rejects. However, here you're returning what it returns, giving it the power to coerce. Which should it be?